Support some conversations api (2)

suzuki-shunsuke opened the pull request:
* https://api.slack.com/methods/conversations.invite
* https://api.slack.com/methods/conversations.kick
* https://api.slack.com/methods/conversations.close
* https://api.slack.com/methods/conversations.create
* https://api.slack.com/methods/conversations.info
* https://api.slack.com/methods/conversations.leave
* https://api.slack.com/methods/conversations.replies
* https://api.slack.com/methods/conversations.list
* https://api.slack.com/methods/conversations.open
* https://api.slack.com/methods/conversations.join
* https://api.slack.com/methods/conversations.history

suzuki-shunsuke committed:
Support conversations.invite API

https://api.slack.com/methods/conversations.invite

suzuki-shunsuke committed:
Support conversations.kick API

https://api.slack.com/methods/conversations.kick

suzuki-shunsuke committed:
Support conversations.close API

https://api.slack.com/methods/conversations.close

suzuki-shunsuke committed:
Support conversations.create API

https://api.slack.com/methods/conversations.create

suzuki-shunsuke committed:
Support conversations.info API

https://api.slack.com/methods/conversations.info

Add `Locale` field to Channel.

suzuki-shunsuke committed:
Support conversations.leave API

https://api.slack.com/methods/conversations.leave

suzuki-shunsuke committed:
Support conversations.replies API

https://api.slack.com/methods/conversations.replies

Add the following fields to the struct `Msg`.

* LastRead
* Subscribed
* UnreadCount

james-lawrence commented:
@suzuki-shunsuke i'll look at this saturday morning. initial impressions look good though.

suzuki-shunsuke committed:
Fix the comment of GetUsersInConversationContext

suzuki-shunsuke committed:
Support conversations.list API #253

https://api.slack.com/methods/conversations.list

Notice that the following response parameters are not supported because their types are ambiguous in the official documentation.

* pending_shared
* previous_names

suzuki-shunsuke committed:
Support conversations.open API #253

https://api.slack.com/methods/conversations.open

suzuki-shunsuke committed:
Support conversations.join API #253

https://api.slack.com/methods/conversations.join

suzuki-shunsuke committed:
Support conversations.history API #253

https://api.slack.com/methods/conversations.history

suzuki-shunsuke committed:
Fix the type of conversation.Priority to float64 #253

```
Unexpected error: %s json: cannot unmarshal number 0.0151970564935 into Go struct field Channel.priority of type int
```

james-lawrence commented on conversation.go:
```diff
@@ -205,3 +221,372 @@ func (api *Client) RenameConversationContext(ctx context.Context, channelID, cha
 }
+
+// InviteUsersToConversation invites users to a channel
+func (api *Client) InviteUsersToConversation(channelID string, users []string) (*Channel, error) {
```
thoughts on changing to `InviteUsersToConversation(channelID string, users ...string)`

james-lawrence commented on conversation.go:
```diff
@@ -205,3 +221,372 @@ func (api *Client) RenameConversationContext(ctx context.Context, channelID, cha
+}
+
+// CloseConversation closes a direct message or multi-person direct message
+func (api *Client) CloseConversation(channelID string) (bool, bool, error) {
```
returns values could use some documentation. consider making then named. `(noop, alreadyClosed bool, err error)`

james-lawrence commented on conversation.go:
```diff
@@ -205,3 +221,372 @@ func (api *Client) RenameConversationContext(ctx context.Context, channelID, cha
+}
+
+// GetConversationReplies retrieves a thread of messages posted to a conversation
+func (api *Client) GetConversationReplies(params *GetConversationRepliesParameters) ([]Message, bool, string, error) {
```
again some documentation over what the return values mean.

james-lawrence commented on conversation.go:
```diff
@@ -205,3 +221,372 @@ func (api *Client) RenameConversationContext(ctx context.Context, channelID, cha
+}
+
+// GetConversationsContext returns the list of channels in a Slack team with a custom context
+func (api *Client) GetConversationsContext(ctx context.Context, params *GetConversationsParameters) ([]Channel, string, error) {
```
return documentation. =)

james-lawrence reviewed (commented):
minor suggestions, will merge once they are addressed.

suzuki-shunsuke committed:
Use named return value and ... parameters

https://github.com/nlopes/slack/pull/252#pullrequestreview-93505362

suzuki-shunsuke commented:
@james-lawrence 
Thank you for reviewing.
I have fixed.

james-lawrence commented:
@suzuki-shunsuke thanks!
